### ISSUE TITLE
ProgressBar with max value 10 can't reach 100%

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#47390]
**Godot version:**

166a6d0 / 3.3rc6

**Steps to reproduce:**
1. Add ProgressBar node
2. Set max value to 10
3.
![image](https://user-images.githubusercontent.com/2223172/112643072-3ef38880-8e44-11eb-9213-f470d4dc26ea.png)
![image](https://user-images.githubusercontent.com/2223172/112643083-43b83c80-8e44-11eb-8022-9625cf05f411.png)
Also the percents are not accurate at all, e.g. 1/10 is 9%. Could be another float precision issue, but it looks really bad in this case.

**Comment by Bogunovich** [comment]
I can only reproduce this bug when running Godot through the .exe, then creating a brand new progress bar. Using a progress bar from a saved project shows the correct percentages.

I can't reproduce the bug when using the visual studio debugger.

**Comment by cobrapitz** [comment]
I think I found the problem, the get_as_ratio() method in progress_bar.cpp returns 0.99999525... 
![image](https://user-images.githubusercontent.com/12397702/112681602-4419fd00-8e6f-11eb-97ba-7b1e613e1478.png)
the ratio then gets converted to an int
![image](https://user-images.githubusercontent.com/12397702/112681817-817e8a80-8e6f-11eb-8f4c-ac9c62aab4ce.png)
which is 99.999999 to int is 99

**Comment by KoBeWi** [comment]
Maybe the value should be rounded.

**Comment by megalobyte** [comment]
@KoBeWi I think rounding should be implemented, but I think there's another issue that needs to be solved. I discovered this bug while investigating this issue:
![precision_error](https://user-images.githubusercontent.com/80655527/112714621-5b211500-8e98-11eb-86a3-ff821618ebd5.png)

I was able to trace the issue down to `editor_properties.cpp:3284`, `EditorInspectorDefaultPlugin::parse_property`. 
![Screenshot from 2021-03-27 01-08-17](https://user-images.githubusercontent.com/80655527/112714732-f914df80-8e98-11eb-84d5-41ad1d16b46f.png)

Despite the name, I believe the `default_float_step` is converting a double into a float. The end result is a step size of 0.00100005. I'm working on submitting a PR that does fix this particular issue, but it doesn't use rounding
